boards/arm/s32z270dc2_r52/board.c

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2022 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/init.h>
+
+#include <OsIf.h>
+
+#ifdef CONFIG_INIT_CLOCK_AT_BOOT_TIME
+#include <Clock_Ip.h>
+#include <Clock_Ip_Cfg.h>
+#endif
+
+static int s32_board_init(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	OsIf_Init(NULL);
+
+#ifdef CONFIG_INIT_CLOCK_AT_BOOT_TIME
+	Clock_Ip_Init(Clock_Ip_aClockConfig);
+#endif
+
+	return 0;
+}
```

Comment:
why do we need board specific clock config? Can't this be handled by SoC code?

Reply:
You are right. Moved to SoC initialization code.